docs/ocean/design_docs/vert-lagrang-remap/vert-lagrang-remap.rst

```diff
@@ -0,0 +1,524 @@
+                        (layerThickness(k,tlev=2) - layerThickness(k,tlev=1))/dt
+
+If `vertVelocityTop` is computed after remapping, then :code:`div_hu` is no
+longer appropriate as it has been remapped. In this case, the Lagrangian layer 
+thickness should be stored in a scratch variable and then the vertical velocity 
+through the top of the layer can be computed:
+
+.. code::
+
+   layerThicknessALE = layerThickness(tlev=2)
+   
+   layerThickness(tlev=2) = layerThicknessTarget
+   
+   vertVelocityTop = (layerThickness(tlev=2) - layerThicknessALE)/dt
+
+The computation of :code:`vertTransportVelocityTop` and 
+:code:`vertGMBolusVelocitytop` is unchanged as these fields represent Eulerian 
+velocities.
+
+Implementation: Grid is sufficiently conditioned for accuracy and stability
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+After determining the target grid, perform the following steps:
+
+#. Optional: Assign :math:`h_k^{t+1}` to :math:`h_k^{lg}` if 
+   :math:`h_k^{t+1} - h_k^{lg}` is less than a minimum thickness alteration. 
+   This motivated by accuracy considerations, as each remapping may introduce 
+   errors. *Darren, would this improve PPR computational performance?*
+#. Apply minimum layer thickness criterion. 
+
+Smoothing layers in space and time is left for a future design document in 
+which we implement support for additional coordinate systems including hybrid 
+coordinates.
+
+Namelist options:
+
+- Minimum layer thickness
+- Optional: minimum thickness change for remapping to occur
+
+
+Implementation: Ability to specify remapping method and its options
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+Namelist options:
+
+- frequency with which remapping should be performed (on which timestep)
+- order of the remapping
+- order of edge slope estimates
+- monotone limiter
+- boundary condition option
+- option to output some diagnostics?
+- *Some other remapping options here*
+
+
+Implementation: Support for existing time stepping schemes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+:code:`vertAleTransportTop` is set to zero for both time stepping schemes from 
+:code:`ocn_vert_transport_velocity_top`.
+
+*Some details here about how to treat ALE_thickness*
+
+Implementation: performance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+Options for improving performance:
+
+- Using the split-explicit scheme
+- Splitting the scalar and momentum timesteps
+- Only remapping when the change in thickness exceeds given threshold
+- Optimizing/parallelizing PPR?
```

Comment:
I would think all of these should be noted as avenues for exploration in a future project but would not be in the scope of the current design.  Instead, I would think our fallback for now if performance isn't great would be that VLR would only be used in contexts where it is required to support a new vertical coordinate, and where the performance hit would be deemed acceptable.